Add network related configs, sort and cleanup config list

LaakkonenJussi opened the pull request:
Add some network related configs based on cross referring between X10 II and III kernel configs. Mainly to ensure that we would not miss some important ones in the future.

piggz commented:
Perhaps add the following notes:
CONFIG_NF_CONNTRACK_IPV4 exists up to 4.18
CONFIG_NF_CONNTRACK_IPV6 exists up to 4.18
CONFIG_NF_NAT_IPV6 exists up to 5.0

This possibly isnt exhaustive, just the ones i couldnt clear on 6.2 on the pinephone pro

LaakkonenJussi commented:
> > Thanks. Added these in [32b6150](https://github.com/mer-hybris/mer-kernel-check/commit/32b615085bc8db21ced1419f8432be41798b4fea)
> 
> The version conditional can be added right after the other conditional options like this "CONFIG_NF_CONNTRACK_IPV4 y,m,!,<=4.18 # optional"

Corrected this and sorting with cdd955498396b07c93bc82c921673b0d7df660c0

LaakkonenJussi commented:
Rearranged and squashed commits. @mlehtima I guess you'd need to merge the changes as I have no permissions to do that.
